src/plone/volto/indexers.py

```diff
@@ -2,6 +2,15 @@
 from plone.dexterity.interfaces import IDexterityContent
 from plone.indexer.decorator import indexer
 from plone.volto.behaviors.preview import IPreview
+from plone.volto.interfaces import IFolderishDocument
+
+
+@indexer(IFolderishDocument)
+def numberOfFolderishDocuments(obj):
```

Comment:
@jaroel I think this metadata will not be updated if we add a content in the container. Indexing would have to be called in the event of adding content. I saw a discussion about this at: https://community.plone.org/t/indexing-how-to-index-contained-items/17428/18

Reply:
I'm adding an indexer, not an event listener. This works just fine (tm).
Thanks for having a look though!

Reply:
@jaroel surprisingly for me it works! I saw that when we add content to a container, it is reindexed. Even its modification date is changed.

@mauritsvanrees @jensens @davisagli , was that thought or is it an "accident"? Indexing is one of the most expensive things in Plone. If this has not been thought of, it would be good to remove this behavior.

Reply:
@wesleybl I don't think that is intentional. Unfortunately changing it could be disruptive since existing implementations are probably relying on it happening. So I would say we can change it in Plone 7 (but maybe there is a way to make it available in an opt-in fashion in Plone 6)

Reply:
I think it may be intentional.  Think only of caching: if content is added to a container, but this does not mark the container as changed, then you will keep getting the previous version, and not see the new content.

Then again, the new content itself get indexed and causes the catalogCounter to be increased, which is usually checked in eTags by plone.app.caching.  So it may not be needed for that.